content/client-lib-development-guide/features.textile

```diff
@@ -232,6 +232,19 @@ h3(#rest-auth). Auth
 ** @(RSA16b)@ If the library is provided with a token without the corresponding @TokenDetails@, then this holds a @TokenDetails@ instance in which only the @token@ attribute is populated with that token string
 ** @(RSA16c)@ Is set with the current token (if applicable) on instantiation and each time it is replaced, whether the result of an explicit @Auth#authorize@ operation, or a library-initiated renewal resulting from expiry or a token error response
 ** @(RSA16d)@ Is empty if there is no current token, including after a previous token has been determined to be invalid or expired, or if the library is using basic auth
+* @(RSA17)@ @Auth#revokeTokens@ function: This function is intended to be used to revoke tokens using provided @TokenRevocationRequest@ details. It internally creates request for  "REST revokeTokens endpoint":/rest-api/#revokeTokens and handles the response.
+** @(RSA17a)@ Method signature is @revokeTokens(TokenRevocationRequest)@.
```

Comment:
I thought that while two of parameters are optional, it could be a waste of parameter space to move these as parameters of the function. Also change on parameters in the future would cause the function interface to change. 
However I think this is a subjective matter and an approach will suit better in a language than other. I see the benefit of using function params, that the passed object will be un-encapsulated and more reachable.  I have used my OOP instincts when writing this. Can I get some opinions about this please @QuintinWillison ?

Reply:
I agree with @SimonWoolf, this makes it harder for the user, they have to create an object, propagage the object, then pass that object to the method, instead of just calling a method.  It's unnecessary indirection.

Reply:
I changed function signature and removed the request object here. 3cc1dab873bfbf3ddf9f3209235fdc0720bafda5